README.md

```diff
@@ -1,43 +1,74 @@
+
+This repository hosts the API defintion of the Custom Resource Definitions (CRD) used for the Container Object Storage Interface (COSI) project. The provisioned unit of storage is a `Bucket`. The following CRDs are defined for managing the lifecycle of Buckets:
+
+ - BucketRequest - Represents a request to provision a Bucket
+ - BucketClass - Represents a class of Buckets with similar characteristics
+ - Bucket - Represents a Bucket or its equivalent in the storage backend
  
-COSI deprecated repos:
+ The following CRDs are defined for managing the lifecycle of workloads accessing the Bucket: 
+ 
+ - BucketAccessRequest - Represents a request to access a Bucket
+ - BucketAccessClass - Represents a class of accesors with similar access requirements
+ - BucketAccess - Represents a access token or service account in the storage backend 
 
-[Spec](https://github.com/container-object-storage-interface/spec) \
-[API](https://github.com/container-object-storage-interface/api) \
-[Manager (i.e. Controller)](https://github.com/container-object-storage-interface/cosi-controller-manager) \
-[Provisioner Sidecar](https://github.com/container-object-storage-interface/cosi-provisioner-sidecar) \
-[Ephemeral CSI Driver](https://github.com/container-object-storage-interface/ephemeral-csi-driver) 
+**NOTE**: All of the APIs are defined under the API group `objectstorage.k8s.io`.
 
+For more information about COSI, visit our [documentation](https://github.com/kubernetes-sigs/container-object-storage-interface-api/tree/master/docs/index.md).
 
-# Container Object Storage Interface (COSI) API
+## Developer Guide
 
-API definitions for Container Object Storage.
+All API definitions are in [`apis/objectstorage.k8s.io/` ](./apis/objectstorage.k8s.io/). All API changes **_MUST_** satisfy the following requirements:
 
-## Community, discussion, contribution, and support
+ - Must be backwards compatible
+ - Must be in-sync with the API definitions in [sigs.k8s.io/container-object-storage-interface-spec](https://github.com/kubernetes-sigs/container-object-storage-interface-spec)
+ 
+### Build and Test
+
+1. Test and Build the project
 
-If you are new to a SIG Storage project, check out this [contributing guide](https://github.com/kubernetes/community/blob/master/sig-storage/CONTRIBUTING.md) 
-and the [community page](https://github.com/kubernetes/community/tree/master/sig-storage). 
+```
+make all
+```
 
-If you are new to the SIG Storage COSI project, check out the [spec](https://github.com/kubernetes-sigs/container-object-storage-interface-spec/blob/master/spec.md), [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support), and [project board](https://github.com/orgs/kubernetes-sigs/projects/).
+2. Generate CRDs
 
-Learn how to engage with the Kubernetes community on the [community page](http://kubernetes.io/community/).
+```
+make codegen
+```
+
+## Adding new fields to protocols
```

Comment:
We should make this an issue template in the repository.

Reply:
I agree, template is already in place.